### PR TITLE
Add PresentEvent grpc message

### PR DESCRIPTION
**Pull request by pierricgimmig**
"PresentEvent" objects represent events related to swap chain presentation 
on Windows. They will be generated by the GraphicsEtwProvider, see #3790 
for more context.

Tests: compilation, end-to-end feature works in #3790

### CODE REVIEW

File: src/GrpcProtos/capture.proto [hunk 1]
```diff
@@ -517,6 +517,23 @@ message GpuDebugMarker {
...
 }
 
+message PresentEvent {
+  uint32 pid = 1;
+  uint32 tid = 2;
+
+  // Unlike other events, we choose the (start_timestamp_ns, duration_ns) pair
+  // as some present "start" events don't have a corresponding "stop".
+  uint64 begin_timestamp_ns = 3;
+  uint64 duration_ns = 4;
```
Comment (ronaldfw) [inline src/GrpcProtos/capture.proto:527]: What does duration for these events mean? Why is this not a single point in time event?

Comment (akopich) [reply src/GrpcProtos/capture.proto:527]: Perhaps, `duration_ns` could help identify "dropped" frames?

File: src/GrpcProtos/capture.proto [hunk 2]
```diff
@@ -517,6 +517,23 @@ message GpuDebugMarker {
   Color color = 5;
 }
 
+message PresentEvent {
```
Comment (akopich) [inline src/GrpcProtos/capture.proto:520]: How will we group the events? pid+tid+source? I would expect smth like `function_id` (e.g. `etw_id`). 

I mean, there might be multiple ETW events, and only some of them will be related to frame track, right?

File: src/CaptureClient/ApiEventProcessorTest.cpp [hunk 3]
```diff
@@ -61,6 +61,7 @@ class MockCaptureListener : public CaptureListener {
               (uint64_t /*timestamp_ns*/,
                std::vector<orbit_grpc_protos::ModuleInfo> /*module_infos*/),
               (override));
+  MOCK_METHOD(void, OnPresentEvent, (const orbit_grpc_protos::PresentEvent&), (override));
```
Comment (akopich) [inline src/CaptureClient/ApiEventProcessorTest.cpp:64]: Shouldn't the call to this method be ensured by EXPECT_CALL in one of the tests?

**Review by akopich — COMMENTED** [review]
LGTM. Some minor questions. Thanks!

**Review by akopich — APPROVED** [review]

File: src/GrpcProtos/capture.proto (hunk 1 above)
Comment (pierricgimmig) [reply src/GrpcProtos/capture.proto:527]: D3d9 and Dxgi emit "PresentStart" and "PresentStop" ETW events. While it is not yet entirely clear what the "PresentStop" represent, the plan is for "duration_ns" to represent the time interval between the start/stop pair.

File: src/GrpcProtos/capture.proto (hunk 2 above)
Comment (pierricgimmig) [reply src/GrpcProtos/capture.proto:520]: The PresentEvent only pertains to graphics ETW related to swapchain presentation, this is not a generic EtwEvent class. As such, those events will be grouped by their "Source", which currently can either be "D3d9" or "Dxgi". All received "PresentEvent" will be displayed on the frame track (well in some form, i.e. the interval between consecutive events).

File: src/CaptureClient/ApiEventProcessorTest.cpp (hunk 3 above)
Comment (pierricgimmig) [reply src/CaptureClient/ApiEventProcessorTest.cpp:64]: Yes good point, added a test in `CaptureEventProcessorTest.cpp`

File: src/GrpcProtos/capture.proto [hunk 4]
```diff
@@ -749,7 +766,7 @@ message ClientCaptureEvent {
     // use them for high frequency events. For the rest please assign
     // numbers starting with 16.
     //
-    // Next high-frequency ID: 12
+    // Next high-frequency ID: 13
```
Comment (dpallotti) [inline src/GrpcProtos/capture.proto:769]: Do I get it right that this event should happen with the same frequency as frame rate? If yes, that's very low frequency, so could you change to use one of the "low-frequency" ids before the branch cut?

File: src/GrpcProtos/capture.proto [hunk 5]
```diff
@@ -807,7 +825,7 @@ message ProducerCaptureEvent {
     // use them for high frequency events. For the rest please assign
     // numbers starting with 16.
     //
-    // Next high-frequency ID: 15
+    // Next high-frequency ID: No high-frequency index left.
```
Comment (dpallotti) [inline src/GrpcProtos/capture.proto:828]: Same here please.

**Pull request merged**